src/libs/ReportUtils.js

```diff
@@ -1250,6 +1250,50 @@ function buildOptimisticAddCommentReportAction(text, file) {
     };
 }
 
+/**
+ * update optimistic parent reportAction when a comment is added or remove in the child report
+ * @param {String} parentReportAction - Parent report action of the child report
+ * @param {String} lastVisibleActionCreated - Last visible action created of the child report
+ * @param {String} type - The type of action in the child report
+ * @returns {Object}
+ */
+
+function updateOptimisticParentReportAction(parentReportAction, lastVisibleActionCreated, type) {
+    let childVisibleActionCount = parentReportAction.childVisibleActionCount || 0;
+    let childCommenterCount = parentReportAction.childCommenterCount || 0;
+    let childOldestFourAccountIDs = parentReportAction.childOldestFourAccountIDs;
+
+    if (type === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD) {
+        childVisibleActionCount += 1;
+        const oldestFourAccountIDs = childOldestFourAccountIDs ? childOldestFourAccountIDs.split(',') : [];
+        if (oldestFourAccountIDs.length < 4) {
+            const index = _.findIndex(oldestFourAccountIDs, (accountID) => accountID === currentUserAccountID.toString());
+            if (index === -1) {
+                childCommenterCount += 1;
+                oldestFourAccountIDs.push(currentUserAccountID);
+            }
+        }
+
+        if (!childCommenterCount) childCommenterCount = 1;
+        if (!oldestFourAccountIDs.length) oldestFourAccountIDs.push(currentUserAccountID);
```

Comment:
So we are only adding the user to the list if nobody else has commented? If there are 4 already, we should be pushing out the oldest and adding the user

Reply:
> or childOldestFourAccountIDs are 4 already, BE will not push the user into the list.

Oh because it's the _oldest_ 4 not the newest 4. Got it.